### DISCUSSION TITLE
Is there any way to get TRPC-like CTRL-Click (CMD-Click) behavior to jump to the corresponding back end procedure from the front end?

### DISCUSSION THREAD

**Discussion by adhanji8** [honojs/hono#2659] [Q&A]
One of the things I like about trpc is the ability to jump to the procedure that corresponds to a call made from the front end. I'm using Hono RPC feature together with React and it's wonderful...but just wondering if this is possible?

In other words, in my front end, if I make a hono rpc call like so:
<img width="765" alt="image" src="https://github.com/honojs/hono/assets/55081439/ca4a0c56-39e9-4cee-983a-e3e423e8696d">

Is there a way I can CMD-click (or CTRL-click) on totalspent and get sent to the corresponding backend endpoint:
<img width="819" alt="image" src="https://github.com/honojs/hono/assets/55081439/6d33c6d5-569c-462a-95be-67d432e2e308">

**Comment by ghost** [comment]
There is currently no solution that I have found.
The likely cause is the internal use of Proxy.

Hono uses [Proxy](https://github.com/honojs/hono/blob/e7098217293a4b45a567de6cb14cde11b6e21e00/src/client/client.ts#L15) and [type utilities](https://github.com/honojs/hono/blob/e7098217293a4b45a567de6cb14cde11b6e21e00/src/client/types.ts#L146) to allow paths defined as strings (e.g., `/posts/create`) to be accessed like objects (e.g., `.posts.create`). As a result, there is no actual object being accessed, so the IDE cannot find its definition source (route).

These issues might be resolved by using methods that don't rely on Proxy and type utilities. We are open to PRs, so feel free to contribute.

**Reply by ghost** [reply]
Do you still have questions or issues? If not, I would like you to close the discussion.

**Reply by adhanji8** [reply]
Got it, thanks!
